Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -110,8 +110,9 @@ RefPtr<ImageBuffer> ImageBitmap::createImageBuffer(ScriptExecutionContext& scrip
 
 void ImageBitmap::createPromise(ScriptExecutionContext& scriptExecutionContext, ImageBitmap::Source&& source, ImageBitmapOptions&& options, ImageBitmap::Promise&& promise)
 {
+    Ref protectedScriptExecutionContext = scriptExecutionContext;
     WTF::switchOn(source,
-        [&] (auto& specificSource) {
+        [scriptExecutionContext = WTFMove(protectedScriptExecutionContext), &options, &promise] (auto& specificSource) {
```

Comment:
This can be one liner: scriptExecutionContext = Ref { protectedScriptExecutionContext }